src/utils/useResizeAware/_ResizeListener.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { MutableRefObject, useCallback, useRef } from 'react';
+
+// This automatically attaches to itself the resize event listener
+// and adds onResize as callback
+export const ResizeListener = ({
+  onResize,
+}: {
+  onResize: (ref: MutableRefObject<HTMLIFrameElement | null>) => void;
+}) => {
+  const ref = useRef<HTMLIFrameElement | null>(null);
```

Comment:
What do you think about simplifying the type? `useRef` infers `null`.
```suggestion
  const ref = useRef<HTMLIFrameElement>(null);
```